source/httpfs/driver_test.go

```diff
@@ -43,59 +48,60 @@ func TestWithInstanceAndNew(t *testing.T) {
 	}
 
 	for _, test := range tests {
-		t.Run("Init "+test.name, func(t *testing.T) {
-			var d httpfs.Driver
+		t.Run(test.name, func(t *testing.T) {
+			var d driver
 			err := d.Init(test.fs, test.path)
```

Comment:
I think it all boils down to visibility.

If we implement `Open()` at the top of the test file and call `Init()` there - tests will look like testing the `Open()` function for correctly calling `Init()` instead of testing `Init()` directly.

Current implementation removes the possibility of incorrect `Open()` implementation failing the tests that are checking for `Init()` correctness.

Furthermore existing `driver` type is used by multiple tests - `TestMigratorInit` and `TestFirstWithNoMigrations`.

Would it be OK to create a new test that implements and uses `Open()` to serve as a reference for new driver implementations?

Reply:
Created `type driverExample struct ` and added `TestDriverExample()`.